src/mscorlib/shared/System/IO/PathInternal.Unix.cs

```diff
@@ -73,6 +78,43 @@ internal static string NormalizeDirectorySeparators(string path)
 
             return builder.ToString();
         }
+
+        internal static ReadOnlySpan<char> NormalizeDirectorySeparators(ReadOnlySpan<char> path)
+        {
+            if (path.IsEmpty) return path;
```

Comment:
you are still duplicating all this code.

Reply:
There is still some functions that need to be optimized... I am working on them